src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocContentLocationCheck.java

```diff
@@ -0,0 +1,240 @@
+ *   *&#47;
+ * &#47;** This comment is OK because it is on the single line. *&#47;
+ *</pre>
+ * <p>
+ * To ensure that Javadoc content starts from the first line:
+ * </p>
+ * <pre>
+ * &lt;module name="JavadocContentLocationCheck"&gt;
+ *   &lt;property name="location" value="FIRST_LINE"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * This setting produces a violation for each comment not
+ * starting on the same line as the initial asterisks:
+ * </p>
+ * <pre>
+ * &#47;** This comment is OK because it starts on the first line.
+ *    * There may be additional text.
+ *    *&#47;
+ * &#47;**
+ *   * This comment causes a violation because it starts on the second line.
+ *   *&#47;
+ * &#47;** This single-line comment also is OK. *&#47;
+ * </pre>
+ *
+ * @since 8.27
+ */
+@StatelessCheck
+public class JavadocContentLocationCheck extends AbstractCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties" file.
+     */
+    public static final String MSG_JAVADOC_CONTENT_FIRST_LINE = "javadoc.content.first.line";
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties" file.
+     */
+    public static final String MSG_JAVADOC_CONTENT_SECOND_LINE = "javadoc.content.second.line";
+
+    /**
+     * Specify the policy on placement of the Javadoc content.
+     */
+    private JavadocContentLocation location = JavadocContentLocation.SECOND_LINE;
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {
+            TokenTypes.BLOCK_COMMENT_BEGIN,
+        };
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public boolean isCommentNodesRequired() {
+        return true;
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        if (isMultilineComment(ast) && JavadocUtil.isJavadocComment(ast)) {
+            final String commentContent = JavadocUtil.getJavadocCommentContent(ast);
+            final int blankLines = countBlankLines(commentContent);
+            if (blankLines >= 0) {
+                if (location == JavadocContentLocation.FIRST_LINE) {
+                    if (blankLines != 0) {
+                        log(ast, MSG_JAVADOC_CONTENT_FIRST_LINE);
+                    }
+                }
+                else if (blankLines != 1) {
+                    log(ast, MSG_JAVADOC_CONTENT_SECOND_LINE);
+                }
+            }
+        }
+    }
+
+    /**
+     * Setter to specify the policy on placement of the Javadoc content.
+     *
+     * @param value string to decode location from
+     * @throws IllegalArgumentException if unable to decode
+     */
+    public void setLocation(String value) {
+        location = JavadocContentLocation.valueOf(value.trim().toUpperCase(Locale.ENGLISH));
```

Comment:
Please put setter above visitToken.
Checks that have special type use special decoder and keep exact enum type in setter.

Reply:
done

Reply:
Second request is not done

Reply:
>  Checks that have special type use special decoder and keep exact enum type in setter.

Please point me to an example of such setter.

Reply:
@romani 
> use special decoder and keep exact enum type in setter

The only way to do this is to put the converter for the enum in the automatic bean ( https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/api/AutomaticBean.java#L165 )
but you told me we couldn't put check specific enumerations in there since it is the API class and you didn't want API to have a direct reliance on a check.

Reply:
Hmm ... I might already forget this. If you know specific comment, please share , lets review.

Check is using specific type is good.  It is not good that we have implementation of AutomaticBean  in api.

Reply:
@pbludov , do you have problems to make spacial type as argument of setter ?
If you want to have String as argument, in this case there is no reason for JavadocContentLocationOption to be public, in should be hidden in Check.

Reply:
Looks like I lost connection with current state of code, all Checks use strings https://github.com/checkstyle/checkstyle/commit/9faf80abaee2a2e224c7d0f9d4a45307b37a398e#diff-9b2a5f9e69b254b923cb4fd7420cb9a4R122 ,  so it is ok to keep code as is in this PR.